packages/components/carousel/src/use-carousel.ts

```diff
@@ -46,6 +46,8 @@ export const useCarousel = (
   const root = ref<HTMLDivElement>()
   const containerHeight = ref<number>(0)
   const isItemsTwoLength = ref(true)
+  const isFirstCall = ref(true)
+  const isTransitioning = ref(false)
```

Comment:
why not use ‘@transitionstart’ instead ‘isFirstCall’

Reply:
I feel using isFirstCall might simplify the code since it eliminates the need for handling event listeners and event handler functions.